src/extension-analytics.js

```diff
@@ -130,17 +130,23 @@ class CustomEventReporter {
  * CustomEventReporter instance.
  */
 export class CustomEventReporterBuilder {
-  /** @param {!AmpElement} parent */
-  constructor(parent) {
+  /**
+   * @param {!AmpElement} parent
+   * @param {?JsonObject} config
+   */
+  constructor(parent, config) {
```

Comment:
instead of exposing whole config option, can we be more specific:

```
setTransportConfig(transportConfig) {
}
```
does this meet your need?